pkg/provider/aws/secret.go

```diff
@@ -2,42 +2,59 @@ package aws
 
 import (
 	"fmt"
-
 	"github.com/klothoplatform/klotho/pkg/core"
-	"github.com/klothoplatform/klotho/pkg/multierr"
 	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
-func (a *AWS) GenerateSecretsResources(construct *core.Secrets, result *core.ConstructGraph, dag *core.ResourceGraph) error {
-	var merr multierr.Error
+func (a *AWS) expandSecrets(dag *core.ResourceGraph, construct *core.Secrets) error {
 	for _, secretName := range construct.Secrets {
-		merr.Append(a.generateSecret(construct, result, dag, secretName))
-	}
-	return merr.ErrOrNil()
-}
-
-func (a *AWS) generateSecret(construct core.Construct, result *core.ConstructGraph, dag *core.ResourceGraph, secretName string) error {
-	secret := resources.NewSecret(construct.Provenance(), secretName, a.Config.AppName)
-	dag.AddResource(secret)
-	a.MapResourceDirectlyToConstruct(secret, construct)
+		secretVersion, err := core.CreateResource[*resources.SecretVersion](dag, resources.SecretVersionCreateParams{
+			AppName:      a.Config.AppName,
+			Refs:         core.AnnotationKeySetOf(construct.AnnotationKey),
+			Name:         secretName,
+			DetectedPath: secretName,
+		})
 
-	secretVersion := resources.NewSecretVersion(secret, secretName)
-	dag.AddDependenciesReflect(secretVersion)
-
-	for _, upstreamCons := range result.GetUpstreamConstructs(construct) {
-		unit, isUnit := upstreamCons.(*core.ExecutionUnit)
-		if !isUnit {
-			continue
+		if err != nil {
+			return err
 		}
 
-		actions := []string{`secretsmanager:DescribeSecret`, `secretsmanager:GetSecretValue`}
-		policyResources := []core.IaCValue{{
-			Resource: secret,
-			Property: resources.ARN_IAC_VALUE,
-		}}
-		policyDoc := resources.CreateAllowPolicyDocument(actions, policyResources)
-		policy := resources.NewIamInlinePolicy(fmt.Sprintf("%s-secretsmanager", secret.Name), core.AnnotationKeySetOf(construct.Provenance()), policyDoc)
-		a.PolicyGenerator.AddInlinePolicyToUnit(unit.Id(), policy)
+		a.MapResourceDirectlyToConstruct(secretVersion.Secret, construct)
 	}
 	return nil
 }
+
+func (a *AWS) getSecretVersionConfiguration(secretVersion *resources.SecretVersion, result *core.ConstructGraph) (resources.SecretVersionConfigureParams, error) {
+	secretVersionConfig := resources.SecretVersionConfigureParams{}
+	if len(secretVersion.ConstructsRef) == 0 {
+		// this case may occur when a secret is created as part of edge expansion and is configured as part of that process
```

Comment:
the version would still have refs about the rds and lambda/exec unit that caused its creation so wouldnt we error here?